spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/support/SimpleTaskRepository.java

```diff
@@ -49,20 +52,31 @@ public SimpleTaskRepository(FactoryBean<TaskExecutionDao> taskExecutionDaoFactor
 	}
 
 	@Override
-	public void update(TaskExecution taskExecution) {
+	public void completeTaskExecution(long executionId, Integer exitCode, Date endTime,
+									  String exitMessage) {
 		initialize();
 
-		validateTaskExecution(taskExecution);
-		taskExecutionDao.updateTaskExecution(taskExecution);
-		logger.info("Updating: " + taskExecution.toString());
+		validateExitInformation(executionId, exitCode, endTime);
+		exitMessage = trimExitMessage(exitMessage);
+		taskExecutionDao.completeTaskExecution(executionId, exitCode, endTime, exitMessage);
+		logger.info("Updating: TaskExecution with executionId="+executionId
```

Comment:
Hmmm...in hind sight we probably should move this to debug to be sure that sensitive data isn't automatically logged.